File: AOJ/0306/symmetric_ternary.java
import java.util.Scanner;

class Main {
  void solve(){
    Scanner sc = new Scanner( System.in );
    int w = sc.nextInt();
    StringBuilder res = new StringBuilder();

    while ( w > 0 ) {
      int r = w % 3;
      if ( r == 0 ) {
        res.insert(0, '0');
        w /= 3;
      } else if ( r == 1 ) {
        res.insert(0, '+');
        w /= 3;
      } else if ( r == 2 ){
        res.insert(0, '-');
        w = w/3 + 1;
      }
    }
	
    System.out.println(res.toString());
  }

  public static void main(String[] args) {
    new Main().solve();
  }
}

File: AOJ/0306/symmetric_ternary.py
w = int(input())
res = ''
while w > 0:
    if w % 3 == 0:
        res += '0'
    elif w % 3 == 1:
        res += '+'
    else:
        res += '-';
        w += 1
    w //= 3
print( res[::-1] )
